drv/sidecar-front-io/src/transceivers.rs

```diff
@@ -77,6 +90,144 @@ impl FpgaPortMasks {
     }
 }
 
+/// Represents a single logical port (0-31)
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub struct LogicalPort(pub u8);
+impl LogicalPort {
+    pub fn as_mask(&self) -> LogicalPortMask {
+        LogicalPortMask(1 << self.0)
+    }
+
+    pub fn get_physical_location(&self) -> PortLocation {
+        PORT_MAP[self.0 as usize]
+    }
+}
+/// Represents a set of selected logical ports, i.e. a 32-bit bitmask
+#[derive(Copy, Clone, Debug, Default, PartialEq)]
+pub struct LogicalPortMask(pub u32);
+
+impl LogicalPortMask {
+    pub const MAX_PORT_INDEX: LogicalPort = LogicalPort(NUM_PORTS - 1);
+
+    pub fn get(&self) -> u32 {
+        self.0
+    }
+    pub fn set(&mut self, index: LogicalPort) {
+        self.0 |= index.as_mask().0
+    }
+    pub fn clear(&mut self, index: LogicalPort) {
+        self.0 &= !index.as_mask().0
+    }
+    pub fn merge(&mut self, other: LogicalPortMask) {
+        self.0 |= other.0
+    }
+    pub fn count(&self) -> usize {
+        self.0.count_ones() as _
+    }
+    pub fn is_set(&self, index: LogicalPort) -> bool {
+        self.0 & index.as_mask().0 != 0
+    }
+    pub fn is_empty(&self) -> bool {
+        self.0 == 0
+    }
+    pub fn to_indices(&self) -> impl Iterator<Item = u8> + '_ {
+        (0..32).filter(|i| self.is_set(LogicalPort(*i)))
+    }
+}
+
+impl From<ModuleId> for LogicalPortMask {
+    fn from(v: ModuleId) -> Self {
+        Self(v.0 as u32)
```

Comment:
`ModuleId` is a u64. We chose something bigger than we needed just to provide a bit of future proofing since this type is going to get baked into the protocol. Since we only support 32 logical ports right now, we can ignore the upper 32 bits.